code/web/src/modules/user/api/action.test.js

```diff
@@ -0,0 +1,59 @@
+import * as actions from './actions.js'
+import axios from 'axios'
+import mockAxios from 'jest-mock-axios';
+jest.mock('axios');
+
+describe('Actions', () => {
+  let user, token;
+  beforeEach(() => {
+    user = {
+      details: {
+        name: "The User",
+        email: "user@crate.com",
+        role: "USER"
+      },
+      error: null,
+      isLoading: false,
+      isAuthenticated: true
+    }
+
+    token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MiwibmFtZSI6IlRoZSBVc2VyIiwiZW1haWwiOiJ1c2VyQGNyYXRlLmNvbSIsInJvbGUiOiJVU0VSIiwiaWF0IjoxNTk4MzMzMTQwfQ.EodZNLw5drqn7FzMTTt4EHujMoAkOeqJaJ5qZXidjho"
+  })
+
+  it('Should set user', () => {
+    const expectedAction = {
+      type: 'AUTH/SET_USER',
+      user
+    }
+    const result = actions.setUser(token, user)
+    expect(result).toEqual(expectedAction)
+  })
+
+  it.skip('Should process user login', async () => {
```

Comment:
Interesting! `.skip` is new to us. What does it do?

Reply:
It skips a test that is either incomplete or not currently passing. In this case, I had some trouble mocking some of the async fn's required for this test to pass so I decided to skip them so I can get to it later while I focus on Layout. 

They are pretty handy! There is one called **.only** that only runs the test suite on that specified test which is helpful when building tests.

Reply:
good to know, thanks!

Reply:
Oh I see, we do have a skip function but it looks a little different. Cool! Glad to know how to recognize it in a javascript test framework!